viabel/objectives.py

```diff
@@ -121,26 +130,148 @@ def __init__(self, approx, model, num_mc_samples, use_path_deriv=False):
             Number of Monte Carlo samples to use to approximate the objective.
         use_path_deriv : `bool`
             Use path derivative (for "sticking the landing") gradient estimator
+        hessian_approx_method : 'string'
+            Select from different methods for approximating the hessian:
+                'full' : use the full hessian matrix provided by BridgeStan
+                'mean_only' : use control variate only for mean estimator to avoid calculation of full hessian
+                'loo_diag_approx' : using "leave one out" method with hessian vector product value at other samples to
+                    estimate the diagonal values of hessian
+                'loo_direct_approx;: the same method as 'loo_diag_approx' but use the scaled approximation to the
+                    gradient of scale to do the "loo" estimation
         """
+
         self._use_path_deriv = use_path_deriv
+        if hessian_approx_method in [None, 'full', 'mean_only', 'loo_diag_approx', 'loo_direct_approx']:
+            self.hessian_approx_method = hessian_approx_method
+        else:
+            raise ValueError("Name of approximation must be one of 'full', 'mean_only', 'loo_diag_approx', "
```

Comment:
Why is this broken into two arguments?

Reply:
Pycharm IDE reported it's too long for a line so it gives an automatic formatting recommendation separating the message into 2 lines. They are functionally equivalent. I can change them into one line.

Reply:
Ah, never mind. I misread and thought there was a comma between the two string arguments. This is find as is.

Reply:
Actually when I was writing the test function to test this validation mechanism, I found that the extra "\n" matters when checking about the equality of 2 strings if we want to check if a correct error message has been reported. They have to be both only in one line or 2 lines, other wise the test fails. Or maybe if we only care about whether the error is raised when input is invalid, we may don't have to check the equality of the message string.

Reply:
The error string shouldn't have a newline in the middle of it.